### PR TITLE
Implemented Loading State

### PR DESCRIPTION
**Pull request by Henrywis**
After realizing that Promise.all will call for a slower render of the fetched API, it seemed to be the best time to implement the Loading state.

<div style="position: relative; padding-bottom: 56.25%; height: 0;"><iframe src="https://www.loom.com/embed/da08a52026014e8b9cef0d7a6f9a3cc2?sid=f42d6817-bcfc-4f87-8125-82b732e3d843" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;"></iframe></div>

### CODE REVIEW

File: iRecruit-ui/src/components/Home/Home.jsx [hunk 1]
```diff
@@ -34,7 +35,7 @@ export default function Home({
   useEffect(() => {
     const interval = setInterval(() => {
       setCarouselIndex((prevIndex) => (prevIndex + 1) % carouselContent.length);
-    }, 7800); // Change carousel content every 5 seconds
+    }, 7800);
```
Comment (9600baud) [inline iRecruit-ui/src/components/Home/Home.jsx:38]: Small suggestion, whenever we have timeout values like this, its useful to show the reader how you got to this number.

For example, if we wanted 5 seconds and the call takes milliseconds we'd setup the constant like this for readability:

`const CAROUSEL_FADE_TIMEOUT = 5 * 1000; // 5 seconds`

File: iRecruit-ui/src/components/Home/Home.jsx [hunk 2]
```diff
@@ -79,37 +80,48 @@ export default function Home({
...
+                    <button onClick={handleCloseApplication}>
+                      Close Application
+                    </button>
+                    <Room
+                      jobId={selectedPostId}
+                      onApplicationSubmit={handleApplicationSubmit}
+                      title={job.title}
+                      joburl={job.url}
+                    />
+                  </>
```
Comment (9600baud) [inline iRecruit-ui/src/components/Home/Home.jsx:114]: This should probably be a separate component or a function that generates this

**Review by 9600baud — COMMENTED** [review]
Looking good

File: iRecruit-ui/src/components/Home/Home.jsx (hunk 1 above)
Comment (Henrywis) [reply iRecruit-ui/src/components/Home/Home.jsx:38]: Oh thank you, initially wanted to use 5 seconds but the fade time and the carousel content display time was not really aligned, so I was basically doing a "line of best fit" trial and error. So I should have taken out the comment.
Also, I will take note of the constant setup.

File: iRecruit-ui/src/components/Home/Home.jsx (hunk 2 above)
Comment (Henrywis) [reply iRecruit-ui/src/components/Home/Home.jsx:114]: Thank you. Please give me more context on what you mean.

**Pull request merged**